include/caffe/solver.hpp

```diff
@@ -129,6 +129,27 @@ class AdaGradSolver : public SGDSolver<Dtype> {
 };
 
 template <typename Dtype>
+class AdaDeltaSolver : public SGDSolver<Dtype> {
+ public:
+  explicit AdaDeltaSolver(const SolverParameter& param)
+      : SGDSolver<Dtype>(param) { PreSolve(); constructor_sanity_check(); }
+  explicit AdaDeltaSolver(const string& param_file)
+      : SGDSolver<Dtype>(param_file) { PreSolve(); constructor_sanity_check(); }
```

Comment:
I suppose you have something wrong here. Now you are calling `PreSolve()` in constructor of both `AdaDeltaSolver` and `SGDSolver`, and since it is a virtual method, you are calling `AdaDeltaSolver::PreSolve()` twice when constructing a AdaDeltaSolver instance. Is that the desired behavior?